serum/_context.py

```diff
@@ -198,7 +201,7 @@ def provide(configuration: DependencyConfiguration) -> Union[T, MagicMock]:
         """
         context = Context.current_context()
 
-        def singleton(singleton_type: Type[T]) -> T:
+        def singleton(singleton_type: T) -> T:
```

Comment:
I think was actually correct as it was: these methods take a type as their argument, and return an instance of that type. But maybe `typing` in 3.5 don't support `Type`? In that case, maybe just remove the type hinting all together.